sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/query.sql

```diff
@@ -201,17 +203,58 @@ client_profile_info AS (
     ANY_VALUE(is_new_profile) AS is_new_profile,
     ANY_VALUE(activity_segment) AS activity_segment
   FROM
-    telemetry_derived.unified_metrics_v1
+    `moz-fx-data-shared-prod.telemetry_derived.unified_metrics_v1`
   WHERE
     submission_date = @submission_date
   GROUP BY
     client_id
+),
+-- Newtab interactions may arrive in different pings so we attach the open details for a visit to all interactions.
+side_filled AS (
+  SELECT
+    * EXCEPT (newtab_open_source, newtab_visit_started_at, newtab_visit_ended_at),
+    FIRST_VALUE(newtab_open_source IGNORE NULLS) OVER (
+      PARTITION BY
+        newtab_visit_id
+      ORDER BY
+        submission_date ASC
+    ) AS newtab_open_source,
+    FIRST_VALUE(newtab_visit_started_at IGNORE NULLS) OVER (
+      PARTITION BY
+        newtab_visit_id
+      ORDER BY
+        submission_date ASC
+    ) AS newtab_visit_started_at,
+    FIRST_VALUE(newtab_visit_ended_at IGNORE NULLS) OVER (
+      PARTITION BY
+        newtab_visit_id
+      ORDER BY
+        submission_date ASC
+    ) AS newtab_visit_ended_at,
+    LOGICAL_OR(
+      searches > 0
+      OR tagged_search_ad_clicks > 0
+      OR tagged_search_ad_impressions > 0
+      OR follow_on_search_ad_clicks > 0
+      OR follow_on_search_ad_impressions > 0
+      OR topsite_impressions > 0
```

Comment:
if we ever add more types of interactions, this list will have to be updated. that's probably OK. in theory another alternative would be to create a catch-all "any_interaction" higher up in the logic, looking for any event_name of `impression`, `click` or `issued`. but that might also be too liberal, if future events have that event_name but we don't want (for some reason) to count it as an interaction. so i guess im just calling it out for now as something we need to be aware of.

Reply:
Good call, I'll add a comment to reflect that